client/hosting/sites/hooks/use-sync-selected-site.ts

```diff
@@ -10,16 +10,9 @@ export function useSyncSelectedSite(
 	selectedSite: SiteDetails | null | undefined
 ) {
 	const dispatch = useDispatch();
-	const isInitialLoad = useRef( true );
 
 	// Update selected site globally as soon as it is clicked from the table.
 	useEffect( () => {
-		// Prevent clearing the selected site on initial load.
-		if ( isInitialLoad.current && ! dataViewsState.selectedItem ) {
-			isInitialLoad.current = false;
-			return;
-		}
-		isInitialLoad.current = false;
 		dispatch( setSelectedSiteId( dataViewsState.selectedItem?.ID ) );
```

Comment:
As I understand, this PR works since `mostRecentlySelectedSiteId` does not get cleared even if this line is executed.